L/libint/build_tarballs.jl

```diff
@@ -0,0 +1,60 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "libint"
+version = v"2.7.0-beta.5"
+
+# Collection of sources required to complete build
+sources = [
+    ArchiveSource("https://github.com/evaleev/libint/releases/download/v2.7.0-beta.5/libint-2.7.0-beta.5.tgz", "38c630d9b4433340079ebc7a93f05d7a285833883327bfc35dd208249e06f05f"),
+    GitSource("https://gitlab.com/libeigen/eigen.git", "21ae2afd4edaa1b69782c67a54182d34efe43f9c"),
+    DirectorySource("./bundled")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+for f in ${WORKSPACE}/srcdir/patches/*.patch; do
+    atomic_patch -p1 ${f}
+done
+cd eigen/
+mkdir build
+cd build/
+cmake -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} ..
+apk add boost-dev
+make install
+cd ../../
+cd libint-2.7.0-beta.5/
+mkdir build
+cd build/
+cmake -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release ..
+cmake --build . --target install -- -j${nprocs}
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
+    Linux(:i686, libc=:glibc),
+    Linux(:x86_64, libc=:glibc),
+    Linux(:aarch64, libc=:glibc),
+    Linux(:armv7l, libc=:glibc, call_abi=:eabihf),
+    Linux(:powerpc64le, libc=:glibc),
+    Linux(:i686, libc=:musl),
+    Linux(:aarch64, libc=:musl),
+    Linux(:armv7l, libc=:musl, call_abi=:eabihf),
+    MacOS(:x86_64)
+]
+
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("liblibint2", :Libint2)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = Dependency[
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version = v"9.1.0")
```

Comment:
Is GCC 9 really required?

Reply:
No, just C++11 support. I figured just choosing the most recent would be the most future proof?

Reply:
Using a recent version of the compiler doesn't help much with future-proofness, on the contrary it can cause huge problems when trying to link to libraries built with older versions of GCC: https://juliapackaging.github.io/BinaryBuilder.jl/dev/troubleshooting/#Building-with-an-old-GCC-version-a-library-that-has-dependencies-built-with-newer-GCC-versions-1 (this doesn't apply to LLVM). Just use the oldest version possible. For C++11 it should be v4.9, or 5, of I remember correctly

Reply:
Good to know! Thanks